ares/js/components/autocomplete/autocomplete.test.js

```diff
@@ -0,0 +1,826 @@
+/* eslint-disable max-classes-per-file */
+import Autocomplete from './autocomplete';
+import Utilities from '../../utilities';
+
+jest.spyOn(Autocomplete.prototype, 'sumbitListener').mockImplementation(() => true);
+jest.spyOn(Autocomplete.prototype, 'removeAttribute').mockImplementation(() => true);
+jest.spyOn(Autocomplete.prototype, 'insertNewSearchInput').mockImplementation(() => true);
+jest.spyOn(Autocomplete.prototype, 'googleMapsScript').mockImplementation(() => true);
+jest.spyOn(Autocomplete.prototype, 'getEventOriginalParams').mockImplementation(() => true);
+jest.spyOn(Autocomplete.prototype, 'setAttribute').mockImplementation(() => true);
+jest.spyOn(Autocomplete.prototype, 'retreiveDestinationValueToPrePopulateInput').mockImplementation(() => 'Austin, TX');
+
+document.body.innerHTML = `<meta name="originalParams" content="siteid=62309&amp;currency=USD&amp;points=-80.104529|26.114917|Tortuga-Sunset Stage,-80.119458|26.100938|Water Taxi Stop (Tickets Extra$),-80.106137|26.110877|Water Taxi Stop (Tickets Extra$)&amp;cid=ROCK&amp;useMiles=&amp;checkin=11/12/21&amp;pageSize=15&amp;mapSize=13&amp;groupid=43285&amp;radius=5&amp;locationlabel=Tortuga-Main Stage&amp;utm_source=internal&amp;nights=3&amp;propertytypes=Hotel,Motel,Resort,Hostel,Ext. Stay,Boutique,Weekly Rentals&amp;latitude=26.10879170000000&amp;map=&amp;longitude=-80.10643370000000&amp;type=geo&amp;properties=x208368,x378,x2636,x2324,x44621,x24437,x29761,x848867,x3846047,x235230,x10505,x3873763,x269736,x1714083,x13941,x39947"><meta name="siteId" content="60279"><div id="theSearchBox"><span>City Search:</span><input type="search" id="address-input" placeholder="Destination" required="true" value="">`;
```

Comment:
After looking through this whole file, you may consider thinking about how to insert/use randomized text, locations, etc. Or at least get closer to this by adding an array of coordinated details (names of locations, variations of those like shortened state names, lat/long for each of those, etc), and then get the test suite to randomly choose one of those, or even run them all since that will test international locations, domestic locations, and you could add any outliers you may find bugs in in the future to that array so that the test suite ensures testing for those weird places that might crop up as you go along.

Again, this is an iteration idea for this test suite, not something I'd halt the roll-out of this for.